### PR TITLE
Scale adaptive concurrency gradient stat

### PR DESCRIPTION
**Pull request by AbdulRahmanAlHamali**
Commit Message: Scale adaptive concurrency gradient stat to improve its granularity
Additional Description: Gauge metrics are integers. Thus, when `gradient` is reported (which has values between 0.5 and 2), we end up only with 0, 1, or 2, which is of little-to-no value. This PR scales it by a 1000, which will give us values between 500 and 2000.
Risk Level: Minimal, unless users have setup systems that read this stat value and act upon it. This is doubtful though, since this metric has been practically useless.
Testing: 
Docs Changes: Clarified the stat values
Release Notes: Adaptive Concurrency Gradient stat was scaled by 1000 to improve its granularity 
Platform Specific Features: N/A
Fixes https://github.com/envoyproxy/envoy/issues/31695

### CODE REVIEW

File: docs/root/configuration/http/http_filters/adaptive_concurrency_filter.rst [hunk 1]
```diff
@@ -202,7 +202,7 @@ The gradient controller uses the namespace
   rq_blocked, Counter, Total requests that were blocked by the filter.
   min_rtt_calculation_active, Gauge, Set to 1 if the controller is in the process of a minRTT calculation. 0 otherwise.
   concurrency_limit, Gauge, The current concurrency limit.
-  gradient, Gauge, The current gradient value.
+  gradient, Gauge, The current gradient value multiplied by 1000 (values will range between 500 and 2000).
```
Comment (jmarantz) [inline docs/root/configuration/http/http_filters/adaptive_concurrency_filter.rst:205]: can you add this to the release notes also?

**Review by jmarantz — COMMENTED** [review]
lgtm modulo release note.

@tonya11en any comments?

**Review by tonya11en — APPROVED** [review]
LGTM. Thanks for improving this.

File: docs/root/configuration/http/http_filters/adaptive_concurrency_filter.rst (hunk 1 above)
Comment (AbdulRahmanAlHamali) [reply docs/root/configuration/http/http_filters/adaptive_concurrency_filter.rst:205]: added

**Comment by jmarantz** [comment]
Needs format fix and  another main merge.

**Review by jmarantz — APPROVED** [review]

**Comment by jmarantz** [comment]
Needs senior maintainer stamp

**Review by zuercher — APPROVED** [review]

**Pull request merged**